utils/mgmt_system/openstack.py

```diff
@@ -621,3 +639,27 @@ def delete_template(self, template_name):
         template = self._find_template_by_name(template_name)
         template.delete()
         wait_for(lambda: not self.does_template_exist(template_name), num_sec=120, delay=10)
+
+    def stack_get(self, stack_id):
+        return self.stackapi.stacks.get(stack_id)
+
+    def stack_exist(self, stack_name):
+        stacks = [stack for stack in self.stack_get(stack_name)
+            if stack.stack_name == stack_name]
```

Comment:
I mentioned some things along these lines in an email, but it looks like we need to be using stack_id at all times with the heat API. We've got a helpful function (maybe more than one) in the ec2 API that's very similar to this list comprehension; I think it makes sense to break this out into a function (like `_get_stack_id`, for example)